pkg/generators/markers.go

```diff
@@ -293,57 +296,150 @@ func lintArrayIndices(markerComments []string, prefix string) error {
 	return nil
 }
 
+// ExtractCommentTags parses comments for lines of the form:
+//
+//	'marker' + "key=value".
+//
+// Values are optional; "" is the default.  A tag can be specified more than
+// one time and all values are returned.  If the resulting map has an entry for
+// a key, the value (a slice) is guaranteed to have at least 1 element.
+//
+// Similar to version from gengo, but this version support only allows one
+// value per key (preferring explicit array indices), supports raw strings
+// with concatenation, and limits the usable characters allowed in a key
+// (for simpler parsing).
+func extractCommentTags(marker string, lines []string) (map[string]string, error) {
+	allowedKeyCharacterSet := `[:_a-zA-Z0-9\[\]\-]`
+	valueEmpty := regexp.MustCompile(fmt.Sprintf(`^\+%s(%s*)$`, marker, allowedKeyCharacterSet))
+	valueAssign := regexp.MustCompile(fmt.Sprintf(`^\+%s(%s*)=(.*)$`, marker, allowedKeyCharacterSet))
+	valueRawString := regexp.MustCompile(fmt.Sprintf(`^\+%s(%s*)>(.*)$`, marker, allowedKeyCharacterSet))
```

Comment:
The marker can change but we only have a few of them in practice.. but we've got to compile regex on each `extractCommentTags` invocation.  That's unfortunate.  I see we're doing a `strings.HasPrefix(line, marker) ` of each line anyway as a pre-filter.  Could we remove the prefix and then use a regex pattern that doesn't need to contain the marker, allowing us to only compile the regexes once (maybe in a sync.Once to avoid doing it in an init and holding the memory if we don't need it?)

Reply:
great point, ive now factored this out